src/console/MeasurementCollectionToFhir/Processor.cs

```diff
@@ -34,15 +38,50 @@ public class Processor : IEventConsumer
             _templateManager = templateManager;
             _measurementImportService = measurementImportService;
             _logger = logger;
+            _retryPolicy = CreateRetryPolicy(logger);
         }
 
         public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
         {
             EnsureArg.IsNotNull(events);
             EnsureArg.IsNotNull(_templateDefinition);
 
-            var templateContent = _templateManager.GetTemplateAsString(_templateDefinition);
+            await _retryPolicy.ExecuteAsync(async () => await ConsumeAsyncImpl(events, _templateManager.GetTemplateAsString(_templateDefinition)));
+        }
+
+        private async Task ConsumeAsyncImpl(IEnumerable<IEventMessage> events, string templateContent)
+        {
             await _measurementImportService.ProcessEventsAsync(events, templateContent, _logger).ConfigureAwait(false);
         }
+
+        private static AsyncPolicy CreateRetryPolicy(ITelemetryLogger logger)
+        {
+            bool ExceptionRetryableFilter(Exception ee)
+            {
+                logger.LogError(new Exception("Encountered retryable exception", ee));
```

Comment:
>logger.LogError(new Exception("Encountered retryable exception", ee)); [](http://example.com/codeflow?start=16&length=70)

Would suggest just logging the original exception. We only get the first inner exception in the logs and if the original exception is already wrapped we will mis the details.  I like the message though, can we switch the message to info? i.e. "Encountered retryable exception {0}", ee.GetType()

Reply:
Will do